litestar/handlers/base.py

```diff
@@ -214,15 +215,6 @@ def _get_merge_opts(self, others: tuple[Router, ...]) -> dict[str, Any]:
         merge_opts["dto"] = value_or_default(self._dto, merge_opts.get("dto", Empty))
         merge_opts["return_dto"] = value_or_default(self._return_dto, merge_opts.get("return_dto", Empty))
 
-        # due to the way we're traversing over the app layers, the middleware stack is
-        # constructed in the wrong order (handler > application). reversing the order
-        # here is easier than handling it correctly at every intermediary step.
-        #
-        # we only call this if 'others' is non-empty, to ensure we don't change anything
-        # if no layers have been merged (happens in '._with_changes' for example)
-        if others:
-            merge_opts["middleware"] = tuple(reversed(merge_opts["middleware"]))
-
```

Comment:
just for my own comprehension, is this removal "replaced" by the  `check_middleware_constraints(self.middleware)` addition below ?